src/client/App.tsx

```diff
@@ -39,6 +41,17 @@ const PrivateRoute = ({ children, ...props }: RouteProps) => {
   );
 };
 
+const history = createBrowserHistory(),
+  pushReplaceHistory: PushReplaceHistory = {
+    push: (location: Location): void => {
+      history.push(location);
+    },
+    replace: (location: Location): void => {
+      history.replace(location);
+    },
+    location: window.location
+  };
```

Comment:
Its a matter of opinion, but I find this more readable:
```suggestion
const history = createBrowserHistory();
const pushReplaceHistory: PushReplaceHistory = {
  push: (location: Location): void => {
    history.push(location);
  },
  replace: (location: Location): void => {
    history.replace(location);
  },
  location: window.location
};
```